components/automate-ui/src/app/modules/infra-proxy/environment-details/environment-details.component.ts

```diff
@@ -0,0 +1,139 @@
+  styleUrls: ['./environment-details.component.scss']
+})
+
+export class EnvironmentDetailsComponent implements OnInit, OnDestroy {
+  public environment: Environment;
+  public cookbookVersions: CookbookVersion[];
+  public tabValue: EnvironmentTabName = 'cookbookConstraints';
+  public url: string;
+  public conflictErrorEvent = new EventEmitter<boolean>();
+  public modalType: string;
+  public serverId: string;
+  public orgId: string;
+  public name: string;
+  public show = false;
+  public hasCookbookConstraints = false;
+  private isDestroyed = new Subject<boolean>();
+  environmentDetailsLoading = true;
+
+  public attributes = new EnvironmentAttributes({
+    default_attributes: '',
+    override_attributes: ''
+  });
+
+  public selectedAttrs: any;
+  public selected_level = 'all';
+
+  // precedence levels
+  public default_attributes = 'default_attributes';
+  public override_attributes = 'override_attributes';
+  public all = 'all';
+
+  @ViewChild(JsonTreeTable, { static: true })
+  tree: JsonTreeTable;
+
+  constructor(
+    private store: Store<NgrxStateAtom>,
+    private router: Router,
+    private layoutFacade: LayoutFacadeService
+  ) { }
+
+  ngOnInit() {
+    this.layoutFacade.showSidebar(Sidebar.Infrastructure);
+    this.store.select(routeURL).pipe(takeUntil(this.isDestroyed))
+      .subscribe((url: string) => {
+        this.url = url;
+        const [, fragment] = url.split('#');
+        this.tabValue = (fragment === 'attributes') ? 'attributes' : 'cookbookConstraints';
+      });
+
+    combineLatest([
+      this.store.select(routeParams).pipe(pluck('id'), filter(identity)),
+      this.store.select(routeParams).pipe(pluck('orgid'), filter(identity)),
+      this.store.select(routeParams).pipe(pluck('name'), filter(identity))
+    ]).pipe(
+      takeUntil(this.isDestroyed)
+    ).subscribe(([server_id, org_id, name]: string[]) => {
+      this.serverId = server_id;
+      this.orgId = org_id;
+      this.name = name;
+      this.store.dispatch(new GetEnvironment({
+        server_id: server_id, org_id: org_id, name: name
+      }));
+    });
+
+    this.store.select(environmentFromRoute).pipe(
+      filter(identity),
+      takeUntil(this.isDestroyed)
+    ).subscribe(environment => {
+      this.show = true;
+      this.environment = environment;
+      this.cookbookVersions = environment.cookbook_versions.filter(Boolean);
+      if (this.cookbookVersions.length > 0) {
+        this.hasCookbookConstraints = true;
+      }
+      this.attributes = new EnvironmentAttributes(this.environment);
+
+      setTimeout(() => this.filter(this.selected_level), 10);
+      this.environmentDetailsLoading = false;
+    });
+  }
+
+  // set selected item to selected_level
+  filter(precedence_level: string): void {
+    this.selected_level = precedence_level;
+    this.selectedAttrs = this.retrieve(precedence_level);
+  }
+
+  // retrieve attributes based on their level of precedence
+  retrieve(level: string): Object {
```

Comment:
@msorens I also changed this to `object` because our default precedence_level is `all`